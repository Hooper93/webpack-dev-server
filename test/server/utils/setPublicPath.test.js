'use strict';

const webpack = require('webpack');
const setPublicPath = require('../../../lib/utils/setPublicPath');

describe('setPublicPath', () => {
  const cases = [
    {
      title: 'no devServer.publicPath, no compiler publicPath',
      webpackPublicPath: null,
      multiCompiler: false,
      options: {},
    },
    {
      title: 'no devServer.publicPath, relative compiler publicPath',
      webpackPublicPath: '/assets/',
      multiCompiler: false,
      options: {},
    },
    {
      title: 'no devServer.publicPath, bad relative compiler publicPath',
      webpackPublicPath: 'assets/',
      multiCompiler: false,
      options: {},
    },
    {
      title: 'no devServer.publicPath, / compiler publicPath',
      webpackPublicPath: '/',
      multiCompiler: false,
      options: {},
    },
    {
      title: 'no devServer.publicPath, absolute url compiler publicPath',
      webpackPublicPath: 'http://localhost:8080/assets/',
      multiCompiler: false,
      options: {},
    },
    {
      title: 'multi compiler, no devServer.publicPath, no compiler publicPath',
      webpackPublicPath: null,
      multiCompiler: false,
      options: {},
    },
    {
      title:
        'multi compiler, no devServer.publicPath, relative compiler publicPath',
      webpackPublicPath: '/assets/',
      multiCompiler: false,
      options: {},
    },
    {
      title:
        'multi compiler, no devServer.publicPath, bad relative compiler publicPath',
      webpackPublicPath: 'assets/',
      multiCompiler: false,
      options: {},
    },
    {
      title: 'multi compiler, no devServer.publicPath, / compiler publicPath',
      webpackPublicPath: '/',
      multiCompiler: false,
      options: {},
    },
    {
      title:
        'multi compiler, no devServer.publicPath, absolute url compiler publicPath',
      webpackPublicPath: 'http://localhost:8080/assets/',
      multiCompiler: false,
      options: {},
    },
    {
      title: 'relative devServer.publicPath',
      webpackPublicPath: '/',
      multiCompiler: false,
      options: {
        publicPath: '/assets/',
      },
    },
    {
      title: 'bad relative devServer.publicPath',
      webpackPublicPath: '/',
      multiCompiler: false,
      options: {
        // this should not be corrected to /assets/, because it should only be fixed
        // if inferred using the compiler options for backwards compatability
        publicPath: 'assets/',
      },
    },
    {
      title: 'absolute devServer.publicPath',
      webpackPublicPath: '/',
      multiCompiler: false,
      options: {
        publicPath: 'http://localhost:8080/assets/',
      },
    },
  ];

  cases.forEach((data) => {
    describe(data.title, () => {
      let compiler;
      beforeAll(() => {
        let webpackConfig;
        if (data.multiCompiler) {
          // eslint-disable-next-line global-require
          webpackConfig = require('../../fixtures/multi-compiler-config/webpack.config');
          if (data.webpackPublicPath) {
            webpackConfig[0].output.publicPath = data.webpackPublicPath;
          }
        } else {
          // eslint-disable-next-line global-require
          webpackConfig = require('../../fixtures/simple-config/webpack.config');
          if (data.webpackPublicPath) {
            webpackConfig.output.publicPath = data.webpackPublicPath;
          }
        }

        compiler = webpack(webpackConfig);
      });

      it('should set correct options.publicPath', () => {
        setPublicPath(compiler, data.options);
        expect(data.options).toMatchSnapshot();
      });
    });
  });
});
