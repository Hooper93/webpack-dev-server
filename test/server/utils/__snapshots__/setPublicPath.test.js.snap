// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`setPublicPath absolute devServer.publicPath should set correct options.publicPath 1`] = `
Object {
  "publicPath": "http://localhost:8080/assets/",
}
`;

exports[`setPublicPath bad relative devServer.publicPath should set correct options.publicPath 1`] = `
Object {
  "publicPath": "assets/",
}
`;

exports[`setPublicPath multi compiler, no devServer.publicPath, / compiler publicPath should set correct options.publicPath 1`] = `
Object {
  "publicPath": "/",
}
`;

exports[`setPublicPath multi compiler, no devServer.publicPath, absolute url compiler publicPath should set correct options.publicPath 1`] = `
Object {
  "publicPath": "http://localhost:8080/assets/",
}
`;

exports[`setPublicPath multi compiler, no devServer.publicPath, bad relative compiler publicPath should set correct options.publicPath 1`] = `
Object {
  "publicPath": "/assets/",
}
`;

exports[`setPublicPath multi compiler, no devServer.publicPath, no compiler publicPath should set correct options.publicPath 1`] = `
Object {
  "publicPath": "http://localhost:8080/assets/",
}
`;

exports[`setPublicPath multi compiler, no devServer.publicPath, relative compiler publicPath should set correct options.publicPath 1`] = `
Object {
  "publicPath": "/assets/",
}
`;

exports[`setPublicPath no devServer.publicPath, / compiler publicPath should set correct options.publicPath 1`] = `
Object {
  "publicPath": "/",
}
`;

exports[`setPublicPath no devServer.publicPath, absolute url compiler publicPath should set correct options.publicPath 1`] = `
Object {
  "publicPath": "http://localhost:8080/assets/",
}
`;

exports[`setPublicPath no devServer.publicPath, bad relative compiler publicPath should set correct options.publicPath 1`] = `
Object {
  "publicPath": "/assets/",
}
`;

exports[`setPublicPath no devServer.publicPath, no compiler publicPath should set correct options.publicPath 1`] = `
Object {
  "publicPath": "/",
}
`;

exports[`setPublicPath no devServer.publicPath, relative compiler publicPath should set correct options.publicPath 1`] = `
Object {
  "publicPath": "/assets/",
}
`;

exports[`setPublicPath relative devServer.publicPath should set correct options.publicPath 1`] = `
Object {
  "publicPath": "/assets/",
}
`;
